source/Cosmos.System2/Graphics/SVGAIICanvas.cs

```diff
@@ -155,7 +166,12 @@ public override void DrawPoint(Pen aPen, float aX, float aY)
         /// <exception cref="NotImplementedException">Thrown if VMWare SVGA 2 has no rectange copy capability</exception>
         public override void DrawFilledRectangle(Pen aPen, int aX_start, int aY_start, int aWidth, int aHeight)
         {
-            _xSVGADriver.Fill((uint)aX_start, (uint)aY_start, (uint)aWidth, (uint)aHeight, (uint)aPen.Color.ToArgb());
+            var color = aPen.Color.ToArgb();
+
+            for (int i = aY_start; i < aY_start + aHeight; i++)
+            {
+                _xSVGADriver.VideoMemory.Fill(GetPointOffset(aX_start, i) + (int)_xSVGADriver.FrameSize, aWidth, color);
```

Comment:
Is this really quicker than just calling the `_xSVGADriver.Fill` which can use the VMWare driver call to fill the entire area with a single update? Might be better if you update the else case in Fill method with the VideoMemory method.

Reply:
Yes it is because the Fill method in the SVGAII driver can be slow due to some calls to WriteToFifo